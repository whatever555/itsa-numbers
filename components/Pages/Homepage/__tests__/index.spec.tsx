import React from 'react'

import { render } from '@testing-library/react'

import { Calculator, performCalculation } from '..'
import { Homepage } from '../'

describe('Homepage', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Homepage />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('page renders ok with values', () => {
    const { getByTestId } = render(<Homepage numA={0} numB={4} />, {})

    const numAInput = getByTestId('numA') as HTMLInputElement
    const numBInput = getByTestId('numB') as HTMLInputElement
    expect(numAInput.value).toBe('0')
    expect(numBInput.value).toBe('4')
  })
})

describe('Calculator', () => {
  it('renders ok with values', () => {
    const { getByTestId } = render(<Calculator numA={0} numB={4} />, {})

    const numAInput = getByTestId('numA') as HTMLInputElement
    const numBInput = getByTestId('numB') as HTMLInputElement
    const result = getByTestId('result') as HTMLElement
    expect(numAInput.value).toBe('0')
    expect(numBInput.value).toBe('4')
    expect(result.innerHTML).toBe('RESULT: 1')
  })

  it('performCalculation returns correct result', () => {
    expect(performCalculation(-4, 4, 3)).toBe(2)
  })

  it('performCalculation returns correct results when the same value is used for both numbers', () => {
    expect(performCalculation(4, 4, 3)).toBe(0)
  })

  it('performCalculation returns correct results when both values are below zero', () => {
    expect(performCalculation(-12, -4, 3)).toBe(3)
  })

  it('performCalculation returns correct results when the second value is lower', () => {
    expect(performCalculation(-12, 21, 3)).toBe(11)
  })

  it('performCalculation returns correct results when max number is large', () => {
    expect(performCalculation(-2, 23423423421, 3)).toBe(7807807807)
  })
})
