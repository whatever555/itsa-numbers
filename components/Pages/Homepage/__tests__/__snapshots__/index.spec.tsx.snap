// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches snapshot 1`] = `
<DocumentFragment>
  <article>
    <header
      class="sc-hKgILt dPKneK"
    >
      Please input two numbers to determine how many numbers between each is divisible by 3
    </header>
    <section>
      <form>
        <div
          class="sc-bdfBwQ kedaRK"
        >
          <label
            class="sc-dlfnbm kFPgZW"
            for="numA"
          >
            Input A:
            <input
              class="sc-gsTCUz jdoKCV"
              data-testid="numA"
              name="numA"
              placeholder="enter a number"
              type="number"
              value="0"
            />
          </label>
          <label
            class="sc-dlfnbm kFPgZW"
            for="numB"
          >
            Input B:
            <input
              class="sc-gsTCUz jdoKCV"
              data-testid="numB"
              name="numB"
              placeholder="enter a number"
              type="number"
              value="0"
            />
          </label>
        </div>
        <footer>
          <div
            class="sc-eCssSg cpBEtW"
            data-testid="result"
          >
            RESULT: 0
          </div>
        </footer>
      </form>
    </section>
    <section />
  </article>
</DocumentFragment>
`;
